File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.java
class Solution {    
    public String firstPalindrome(String[] words) {
        for (String word : words) {
            if (isPalindrome(word)) {
                return word;
            }
        }
        return "";
    }
    
    private boolean isPalindrome(String word) {
        int n = word.length();
        for (int i = 0; i < n/2; i++) {
            if (!Objects.equals(word.charAt(i), word.charAt(n-i-1))) {
                return false;
            }
        }
        return true;
    }
}

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.py
class Solution:
    def isPalindrome(self, word: str) -> bool:
        n = len(word)
        if n%2==0:
            return word[:n//2] == word[n//2:][::-1]
        return word[:n//2+1] == word[n//2:][::-1]
    
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            if self.isPalindrome(word):
                return word
        return ""
    